subsys/net/ip/net_context.c

```diff
@@ -896,8 +896,13 @@ int net_context_create_ipv4_new(struct net_context *context,
 
 	net_pkt_set_ipv4_ttl(pkt, net_context_get_ipv4_ttl(context));
 #if defined(CONFIG_NET_CONTEXT_DSCP_ECN)
-	net_pkt_set_ip_dscp(pkt, net_ipv4_get_dscp(context->options.dscp_ecn));
-	net_pkt_set_ip_ecn(pkt, net_ipv4_get_ecn(context->options.dscp_ecn));
+	/* Direct priority takes precedence over DSCP */
+	if (!IS_ENABLED(CONFIG_NET_CONTEXT_PRIORITY)) {
+		net_pkt_set_ip_dscp(pkt, net_ipv4_get_dscp(context->options.dscp_ecn));
+		net_pkt_set_ip_ecn(pkt, net_ipv4_get_ecn(context->options.dscp_ecn));
```

Comment:
I wonder, shouldn't DSCP/ECN be set regardless of `CONFIG_NET_CONTEXT_PRIORITY`?